easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -607,11 +608,56 @@ def dependencies(self):
         # each processed easyconfig to remove the unwanted dependencies
         self.log.debug("Dependencies BEFORE filtering: %s" % deps)
         filter_deps = build_option('filter_deps')
+
         if filter_deps:
+            # find if any of the filter_deps are in the form name=version or name=[low:high[
+            complex_fdeps_list = [x for x in filter_deps if "=" in x]
+            complex_fdeps = { }
+            # convert list of k=v into dictionary
+            if complex_fdeps_list:
+                for s in complex_fdeps_list:
+                    k,v = s.split('=')
```

Comment:
don't use single-letter variables (outside of list comprehensions), it makes the code quite hard to digest

I'd go with `dep_name` & `dep_version` here, and `dep_spec` rather than `s`?